segfault from test_terminus.t in Travis-CI

As reported in #3067 and elsewhere `test_terminus.t` very occasionally segfaults in travis. I haven't been able to reproduce this outside of travis, and corefiles don't seem to be working inside travis, so I'm not sure how to resolve this issue. Perhaps have to inspect the test and look for potential errors.

This old issue still reproduces from time to time in Github CI. Recently it occurred in the ASan build, so we got a stack trace:
```
2024-04-16T13:45:06.7899899Z ##[error]ERROR: test_terminus.t - exited with status 1
2024-04-16T13:45:06.7903261Z ##[error]AddressSanitizer: SEGV on unknown address 0x000100000057 (pc 0x7ffff73f784d bp 0x613000000040 sp 0x7fffffffe0e0 T0)
2024-04-16T13:45:06.7905227Z ##[group]./src/common/libterminus/test_terminus.log
2024-04-16T13:45:06.7905753Z AddressSanitizer:DEADLYSIGNAL
2024-04-16T13:45:06.7906097Z =================================================================
2024-04-16T13:45:06.7907044Z ==54495==ERROR: AddressSanitizer: SEGV on unknown address 0x000100000057 (pc 0x7ffff73f784d bp 0x613000000040 sp 0x7fffffffe0e0 T0)
2024-04-16T13:45:06.7908155Z ==54495==The signal is caused by a READ memory access.
2024-04-16T13:45:06.7908629Z     #0 0x7ffff73f784d in getenv (/lib64/libc.so.6+0x4084d)
2024-04-16T13:45:06.7909149Z     #1 0x7ffff779d48b in loop_init /usr/src/src/common/libev/ev.c:3287
2024-04-16T13:45:06.7909728Z     #2 0x7ffff779e323 in ev_loop_new /usr/src/src/common/libev/ev.c:3557
2024-04-16T13:45:06.7910329Z     #3 0x7ffff77b0380 in flux_reactor_create /usr/src/src/common/libflux/reactor.c:75
2024-04-16T13:45:06.7910987Z     #4 0x7ffff77b056e in flux_get_reactor /usr/src/src/common/libflux/reactor.c:101
2024-04-16T13:45:06.7911637Z     #5 0x7ffff77ca9af in flux_future_set_flux /usr/src/src/common/libflux/future.c:406
2024-04-16T13:45:06.7912271Z     #6 0x7ffff77c40a5 in rpc_create /usr/src/src/common/libflux/rpc.c:117
2024-04-16T13:45:06.7912900Z     #7 0x7ffff77c40a5 in flux_rpc_message_send_new /usr/src/src/common/libflux/rpc.c:256
2024-04-16T13:45:06.7913545Z     #8 0x7ffff77c4fd2 in flux_rpc_vpack /usr/src/src/common/libflux/rpc.c:400
2024-04-16T13:45:06.7914137Z     #9 0x7ffff77c5245 in flux_rpc_pack /usr/src/src/common/libflux/rpc.c:418
2024-04-16T13:45:06.7914681Z     #10 0x403d4d in test_kill_server_empty test/terminus.c:97
2024-04-16T13:45:06.7915429Z     #11 0x403d4d in main test/terminus.c:480
2024-04-16T13:45:06.7915905Z     #12 0x7ffff73e054f in __libc_start_call_main (/lib64/libc.so.6+0x2954f)
2024-04-16T13:45:06.7916505Z     #13 0x7ffff73e0608 in __libc_start_main@GLIBC_2.2.5 (/lib64/libc.so.6+0x29608)
2024-04-16T13:45:06.7917168Z     #14 0x4058a4 in _start (/usr/src/src/common/libterminus/test_terminus.t+0x4058a4)
2024-04-16T13:45:06.7917578Z 
2024-04-16T13:45:06.7917751Z AddressSanitizer can not provide additional info.
2024-04-16T13:45:06.7918266Z SUMMARY: AddressSanitizer: SEGV (/lib64/libc.so.6+0x4084d) in getenv
2024-04-16T13:45:06.7918706Z ==54495==ABORTING
```

--- a/src/common/libterminus/test/terminus.c
+++ b/src/common/libterminus/test/terminus.c
@@ -68,9 +68,6 @@ static int terminus_server (flux_t *h, void *arg)
     int rc = -1;
     struct flux_terminus_server *t;
 
-    /*  set rank == -1 for testing. Avoid flux_get_rank(3) */
-    setenv ("FLUX_TERMINUS_TEST_SERVER", "t", 1);
-
     /*  N.B.: test_server handle `h` already has reactor with SIGCHLD
      *   flag set.
      */
@@ -475,6 +472,9 @@ int main (int argc, char *argv[])
      */
     if (getenv ("SHELL") == NULL)
         setenv ("SHELL", "/bin/sh", 1);
+
+    /*  set rank == -1 for testing. Avoid flux_get_rank(3) */
+    setenv ("FLUX_TERMINUS_TEST_SERVER", "t", 1);
 
     test_invalid_args ();
     test_kill_server_empty ();

